frontend/src/pages/search.tsx

```diff
@@ -87,82 +221,400 @@ export default function DatasetSearch({
         <link rel="icon" href="/favicon.ico" />
       </Head>
       <Layout backgroundEffect effectSize="100px">
-        <div className="container">
+        <div className="container overflow-x-hidden">
           <div className="pt-5">
-            <div className="relative flex w-full items-center rounded-[12px] border border-[#D1D5DB] ">
-              <input
-                className="w-full grow rounded-[12px] border-0 py-[18px] pl-4 pr-[150px] focus:ring-1 focus:ring-[#006064] "
-                placeholder="Find statistics, forecasts & studies"
-              />
-              <span
-                className="absolute right-[125px] z-[10] cursor-pointer p-2 text-gray-500"
-                role="button"
-              >
-                <XMarkIcon width={20} />
-              </span>
-
-              <Button
-                type="submit"
-                className="absolute right-[10px] top-[10px] flex gap-[8px]"
-              >
-                <MagnifyingGlassIcon width={20} />
-                Search
-              </Button>
-            </div>
+            <SearchBar
```

Comment:
Pressing enter in this searchbar does not trigger a search
![image](https://github.com/user-attachments/assets/35272be6-2194-4628-ac61-c2e2d83d5e34)